tests/components/zerproc/test_light.py

```diff
@@ -0,0 +1,316 @@
+            async_fire_time_changed(hass, utcnow)
+            await hass.async_block_till_done()
+
+        with patch.object(mock_light, "set_color") as mock_set_color:
+            await hass.services.async_call(
+                "light",
+                "turn_on",
+                {ATTR_ENTITY_ID: "light.ledblue_ccddeeff", ATTR_HS_COLOR: (50, 50)},
+                blocking=True,
+            )
+            await hass.async_block_till_done()
+
+        mock_set_color.assert_called_with(75, 68, 37)
+
+        with patch.object(mock_light, "set_color") as mock_set_color:
+            await hass.services.async_call(
+                "light",
+                "turn_on",
+                {
+                    ATTR_ENTITY_ID: "light.ledblue_ccddeeff",
+                    ATTR_BRIGHTNESS: 200,
+                    ATTR_HS_COLOR: (75, 75),
+                },
+                blocking=True,
+            )
+            await hass.async_block_till_done()
+
+        mock_set_color.assert_called_with(162, 200, 50)
+
+
+async def test_light_turn_off(hass, mock_light):
+    """Test ZerprocLight turn_on."""
+    with patch.object(mock_light, "turn_off") as mock_turn_off:
+        await hass.services.async_call(
+            "light",
+            "turn_off",
+            {ATTR_ENTITY_ID: "light.ledblue_ccddeeff"},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+    mock_turn_off.assert_called()
+
+
+async def test_light_update(hass, mock_light):
+    """Test ZerprocLight update."""
+    utcnow = dt_util.utcnow()
+
+    state = hass.states.get("light.ledblue_ccddeeff")
+    assert state.state == STATE_OFF
+    assert state.attributes == {
+        ATTR_FRIENDLY_NAME: "LEDBlue-CCDDEEFF",
+        ATTR_SUPPORTED_FEATURES: SUPPORT_BRIGHTNESS | SUPPORT_COLOR,
+    }
+
+    # Make sure no discovery calls are made while we emulate time passing
+    with patch("homeassistant.components.zerproc.light.pyzerproc.discover"):
+        # Test an exception during discovery
+        with patch.object(
+            mock_light, "get_state", side_effect=pyzerproc.ZerprocException("TEST")
+        ):
+            utcnow = utcnow + SCAN_INTERVAL
+            async_fire_time_changed(hass, utcnow)
+            await hass.async_block_till_done()
+
+        state = hass.states.get("light.ledblue_ccddeeff")
+        assert state.state == STATE_UNAVAILABLE
+        assert state.attributes == {
+            ATTR_FRIENDLY_NAME: "LEDBlue-CCDDEEFF",
+            ATTR_SUPPORTED_FEATURES: SUPPORT_BRIGHTNESS | SUPPORT_COLOR,
+        }
+
+        with patch.object(
+            mock_light,
+            "get_state",
+            return_value=pyzerproc.LightState(False, (200, 128, 100)),
+        ):
+            utcnow = utcnow + SCAN_INTERVAL
+            async_fire_time_changed(hass, utcnow)
+            await hass.async_block_till_done()
+
+        state = hass.states.get("light.ledblue_ccddeeff")
+        assert state.state == STATE_OFF
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)